apps/meteor/ee/client/hooks/useOutboundDialer.ts

```diff
@@ -0,0 +1,40 @@
+import { useSafely } from '@rocket.chat/fuselage-hooks';
+import { useEffect, useState } from 'react';
+
+// import { useVoipClient } from '../../../client/providers/CallProvider/hooks/useVoipClient';
+import { useCallClient } from '../../../client/contexts/CallContext';
+import { hasLicense } from '../../app/license/client';
+import { EEVoipClient } from '../lib/voip/EEVoipClient';
+
+type UseOutboundDialerResult = {
+	outboundDialer?: EEVoipClient;
+	error?: Error | unknown;
+};
+
+const empty = {};
+
+export const useOutboundDialer = (): UseOutboundDialerResult => {
```

Comment:
This hook could be opmized by using another hook called `useHasLicence`. Ex.: 
```
const useOutboundDialer = (): UseOutboundDialerResult => {
    const isEE = useHasLicence('voip-enterprise');
    const voipClient = useCallClient()

    useEffect(() => () => voipClient?.clear())
    
    return isEE && voipClient
        ? { outboundDialer: voipClient } 
        : {}
}
```
The code above is just to give the idea and might not be complete.

Reply:
Something to also consider is whether or not is a good idea to restrict the access to the voipClient based on the licence. I feel like this restriction might make more sense elsewhere.

Reply:
@aleksandernsilva , the access to VoIP client is not restricted based on the license. If the system is in CE mode, https://github.com/RocketChat/Rocket.Chat/blob/870a3e0affb14abeb228e99e5f64cb6c5e37bc18/apps/meteor/client/lib/voip/SimpleVoipUser.ts#L29 would create the object of parent VoIPUser. VoIPUser does not provide outbound calling.  All the components common to EE and CE will use hook 'useVoipClient' to get the access to voipClient to receive calls.

Dialpad is suppose to use useOutboundDialer for getting access to EEVoipClient. As mentioned above this gets created in SimpleVoipUser when there is a proper license. Even though useOutboundDialer will be used by EE component, there is a possibility that it might get used by CE component (Either by mistake or with a hacker mind.)
In that case it may make sense to add a license check here. Wdyt?

Reply:
> 

@aleksandernsilva I do not find useHasLicence. So could not change this part of the code.